deepchecks/nlp/utils/text_properties.py

```diff
@@ -715,79 +772,74 @@ def calculate_builtin_properties(
     )
+    import_warnings = set()
 
     progress_bar = create_progress_bar(
-        iterable=list(text_properties),
-        name='Text Properties Calculation',
-        unit='Text Property'
+        iterable=list(raw_text),
+        name='Text Samples Calculation',
+        unit='Text Sample'
     )
-
-    # TODO: refactor
-    for prop in progress_bar:
+    for text in progress_bar:
         progress_bar.set_postfix(
-            {'Property': prop['name']},
+            {'Sample': truncate_string(text, max_length=20) if text else 'EMPTY STRING'},
             refresh=False
         )
-        if prop['name'] not in english_properties_names:
-            try:
-                values = run_available_kwargs(prop['method'], raw_text=raw_text, **kwargs)
-            except ImportError as e:
-                warnings.warn(warning_message.format(prop['name'], str(e)))
-                continue
-            else:
-                calculated_properties[prop['name']] = values
-        else:
-            try:
-                values = run_available_kwargs(prop['method'], raw_text=english_samples, **kwargs)
-            except ImportError as e:
-                warnings.warn(warning_message.format(prop['name'], str(e)))
+        if pd.isna(text):
+            for prop in text_properties:
+                calculated_properties[prop['name']].append(np.nan)
+            continue
+        sample_language = run_available_kwargs(language, text=text, **kwargs)
+        if is_language_property_requested:
+            calculated_properties['Language'].append(sample_language)
+
+        for prop in text_properties:
+            if prop['name'] in import_warnings:  # Skip properties that failed to import:
+                calculated_properties[prop['name']].append(np.nan)
                 continue
+            if sample_language != 'en' and prop['name'] in english_properties_names:
+                calculated_properties[prop['name']].append(np.nan)
             else:
-                result = []
-                idx = 0
-                fill_value = np.nan if prop['output_type'] == 'numeric' else None
-                for mask in english_samples_mask:
-                    if mask:
-                        result.append(values[idx])
-                        idx += 1
-                    else:
-                        result.append(fill_value)
-                calculated_properties[prop['name']] = result
+                try:
+                    value = run_available_kwargs(prop['method'], text=text, **kwargs)
+                except ImportError as e:
+                    warnings.warn(warning_message.format(prop['name'], str(e)))
+                    import_warnings.add(prop['name'])
+                    continue
+                else:
+                    calculated_properties[prop['name']].append(value)
+
+        # Clear property caches:
+        textblob_cache.clear()
+        words_tokens_cache.clear()
+        sentence_tokens_cache.clear()
+    gc.collect()
```

Comment:
What does this do? add comment

Reply:
Cleans RAM from objects without pointers. Added comment